dask/array/_array_expr/_creation.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import annotations
+
+import functools
+from functools import partial
+
+import numpy as np
+
+from dask._collections import new_collection
+from dask.array._array_expr._expr import ArrayExpr
+from dask.array.chunk import arange as _arange
+from dask.array.chunk import linspace as _linspace
+from dask.array.core import normalize_chunks
+from dask.array.utils import meta_from_array
+from dask.array.wrap import _parse_wrap_args, broadcast_trick
+from dask.blockwise import blockwise as core_blockwise
+from dask.layers import ArrayChunkShapeDep
+
+
+class Arange(ArrayExpr):
+    _parameters = ["start", "stop", "step", "chunks", "like", "dtype", "kwargs"]
+    _defaults = {"chunks": "auto", "like": None, "dtype": None, "kwargs": None}
+
+    @functools.cached_property
+    def num_rows(self):
+        return int(max(np.ceil((self.stop - self.start) / self.step), 0))
+
+    @functools.cached_property
+    def dtype(self):
+        return (
+            self.operand("dtype")
+            or np.arange(
+                self.start,
+                self.stop,
+                self.step * self.num_rows if self.num_rows else self.step,
+            ).dtype
+        )
+
+    @functools.cached_property
+    def _meta(self):
+        return meta_from_array(self.like, ndim=1, dtype=self.dtype)
+
+    @functools.cached_property
+    def chunks(self):
+        return normalize_chunks(
+            self.operand("chunks"), (self.num_rows,), dtype=self.dtype
+        )
+
+    def _layer(self) -> dict:
+        dsk = {}
+        elem_count = 0
+        start, step = self.start, self.step
+        like = self.like
+        func = partial(_arange, like=like)
+
+        for i, bs in enumerate(self.chunks[0]):
+            blockstart = start + (elem_count * step)
+            blockstop = start + ((elem_count + bs) * step)
+            task = (
+                func,
+                blockstart,
+                blockstop,
+                step,
+                bs,
+                self.dtype,
+            )
```

Comment:
I suggest to use new-style tasks whenever possible.
```suggestion
            task = Task(
                (self._name, i),
                func,
                blockstart,
                blockstop,
                step,
                bs,
                self.dtype,
            )
```